byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global google */
+
+if(document.readyState === 'loading') {
+  document.addEventListener('DOMContentLoaded', initialize);
+} else {
+  initialize();
+}
+
+let map;
+let placesService;
+let randomLocation;
+const RADIUS = 1000;
+
+/**
+ * initializes the webpage with a map and loads
+ * recommendations based on user interest.
+ */
+function initialize() {
+  randomLocation = new google.maps.LatLng(33.4806, -112.1864);
+  map = new google.maps.Map(document.getElementById('map'), {
+      center: randomLocation,
+      zoom: 15
+  });
+  placesService = new google.maps.places.PlacesService(map);
+  loadRecommendations();
+}
+
+/**
+ * Calls a Servlet to find a user's interests and
+ * uses a text search to find places fitting those
+ * loaded entires.
+ */
+function loadRecommendations() {
+  fetch('/api/generator')
+  .then(response => response.json())
+  .then((interests) => {
+    interests.forEach(placeType => {
+      console.log(placeType);
+      let request = {
+        location: randomLocation,
+        radius: RADIUS,
+        query: placeType
+      }
+      placesService.textSearch(request, addRecommendations);
+    });
+  });
+}
+
+/**
+ * Checks the response from PlacesService and creates markers
+ * on the locations found from the request.
+ * @param {Array results} contains the results of the request.
+ * @param {PlacesServiceStatus status} contains the service status
+ * of the request.
+ */
+function addRecommendations(results, status) {
+  if (status == google.maps.places.PlacesServiceStatus.OK) {
+    let maxRecommendations = 3;
+    let numRecommendations = 0;
+    for (let i = 0; i < results.length; i++) {
```

Comment:
could you loop through results using foreach, so you dont have to use i to access the result?

Reply:
I initially did use `.forEach`, but i also had to use the `i` variable to keep track of the current recommendations loaded to limit the results it showed with `maxRecommendations`, or else it could flood the map with a lot of markers. I had to keep track of the `i` counter, so I just used a standard for loop instead

Reply:
Also, this is a temporary solution, since the user might want to load more recommendations than what we show at first. I'll add that into function's comments and set it aside as a TODO